pkg/operator/controller/gateway-service-dns/controller.go

```diff
@@ -222,6 +223,10 @@ func (r *reconciler) ensureDNSRecordsForGateway(ctx context.Context, gateway *ga
 	var errs []error
 	for _, domain := range domains {
 		name := operatorcontroller.GatewayDNSRecordName(gateway, domain)
+		// If domain doesn't have a trailing dot, add it
```

Comment:
```suggestion
		// If domain doesn't have a trailing dot, add it.
```
I would have added the trailing dot in `desiredDNSRecord`, but maybe this makes more sense.